.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,14 +1,63 @@
-For new packages please briefly describe the package or provide a link to its homepage.
+For package updates, please link to a changelog or describe the upstream changes yourself. This helps your fellow maintainers discover breaking changes which they need to look out for.
+For new packages, please briefly describe what the package is and why it should be added to Nixpkgs (i.e. because it's a popular, new, handy tool). Ideally also provide a link to its homepage.
+
+The following sections are entirely *optional* but most of the information that it asks you to provide will have to be enquired in any case (https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#the-committers-perspective). Filling this out at least a little bit goes a long way towards speeding up the review process, so please try to write as much as you think is reasonable.
+-->
+
+## What was tested
+
+<!--
+Us Nixpkgs maintainers need confidence that your change is good and doesn't break anything unexpectedly. If you have done testing on your changes to verify that they work as intended, please tell us about it. This helps us gain confidence in the quality of your PR which greatly helps to speed up the review process.
+
+I.e. that you have executed the [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests)) and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests).
+
+Or, for functions and "core" functionality, the tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
+
+Also make sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages so that automated tools can find, execute and report them.
+
+(You don't need to write prose, bullet points are sufficient.)
+-->
+
+- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
+- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
+- [x] I have not tested these changes <!-- Please describe your testing! -->
+
+<!--
+Invite other users of the component(s) you changed to review your PR by [@-mentioning](https://github.blog/news-insights/mention-somebody-they-re-notified/) them or requesting a review if they're part of the NixOS Github organisation.
+When other users of the component test your PR and find it to work as expected, it greatly helps reviewers to gain confidence in the quality of your changes.
 -->
 
+
+
+If you actively use the component(s) changed in this PR, please test the component(s) built from this PR and report your findings. It really helps a lot.
```

Comment:
Ideally, we'd have a section in the contributor's manual that describes the basics of how to even test a PR and link it here: How to check out a PR using git, building a package, building Nixpkgs with a local checkout etc.